src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/Marshalling/ByValueContentsMarshalKindValidator.cs

```diff
@@ -44,6 +44,13 @@ private static ResolvedGenerator ValidateByValueMarshalKind(TypePositionInfo inf
             }
             else if (info.ByValueContentsMarshalKind == ByValueContentsMarshalKind.In)
             {
+                if (info.ManagedType is SzArrayType && generator.Generator is StaticPinnableManagedValueMarshaller)
+                {
+                    return ResolvedGenerator.ResolvedWithDiagnostics(s_forwarder, generator.Diagnostics.Add(new GeneratorDiagnostic.NotSupported(info, context)
+                    {
+                        NotSupportedDetails = SR.InAttributeNotSupportedWithoutOutBlittableArray
+                    }));
+                }
```

Comment:
Is there a way we could roll this into the "unnecessary data" support that we recently added instead of hard-coding it here? We could provide a "default" helper that would return "unnecessary" for In-only, and for custom type marshallers that are pinnable, we'd return "not supported" instead.

Reply:
So we want to allow the `[InAttribute]` on most types, just make it unnecessary? And then for pinnable marshallers we'll warn that '[In]' isn't supported?

Reply:
Yes, that would be the best behavior.